src/main/kotlin/app/revanced/patcher/InstructionFilter.kt

```diff
@@ -0,0 +1,313 @@
+package app.revanced.patcher
+
+import app.revanced.patcher.InstructionFilter.Companion.METHOD_MAX_INSTRUCTIONS
+import app.revanced.patcher.extensions.InstructionExtensions.instructions
+import com.android.tools.smali.dexlib2.Opcode
+import com.android.tools.smali.dexlib2.iface.ClassDef
+import com.android.tools.smali.dexlib2.iface.Method
+import com.android.tools.smali.dexlib2.iface.instruction.Instruction
+import com.android.tools.smali.dexlib2.iface.instruction.ReferenceInstruction
+import com.android.tools.smali.dexlib2.iface.instruction.WideLiteralInstruction
+import com.android.tools.smali.dexlib2.iface.reference.FieldReference
+import com.android.tools.smali.dexlib2.iface.reference.MethodReference
+import java.util.EnumSet
+import kotlin.collections.forEach
+
+interface InstructionFilter {
```

Comment:
This class can be put inside Fingerprint.kt since it's closely related.

Reply:
Modern IDEs can navigate to definitions:

![image](https://github.com/user-attachments/assets/7a7a9e0a-ed72-42a0-8b24-4f14b78e1674)